tests/test_utils_reqser.py

```diff
@@ -53,6 +54,7 @@ def _assert_same_request(self, r1, r2):
         self.assertEqual(r1._encoding, r2._encoding)
         self.assertEqual(r1.priority, r2.priority)
         self.assertEqual(r1.dont_filter, r2.dont_filter)
+        self.assertEquals(r1.flags, r2.flags)
```

Comment:
thanks for the patience ,  hope I addressed all the issues you raised please let me know if I missed anything